views.py

```diff
@@ -1,45 +1,91 @@
-            rand_nums = random.sample(range(1, page_len), number)
-            random_pages = []
-            for num in rand_nums:
-                random_pages.append(page_info[num])
-            pages = map(_get_page_by_info, random_pages)
-
-    return render_to_response('featured.html',
-                           dictionary=locals(),
-                           context_instance=RequestContext(request))   
+    # Seed the RNG with today's date so we always feature the same page(s) for
+    # an entire day
+    random.seed(datetime.date.today().strftime("%Y%m%d"))
+
+    # Get and randomize pages
+    pages = []
+    all_pages, this_day_title = _get_pages()
+    if len(all_pages) > 0:
+        rand_nums = random.sample(xrange(len(all_pages)), config.NUMBER)
+        for num in rand_nums:
+            pages.append(all_pages[num])
 
+    # Clear the seed so anything else using random numbers isn't affected
+    random.seed(None)
+
+    return render(request, 'featured.html', locals())
 
 # helper methods
 
-def _get_page_info_by_id(page_id):
-    page_obj = models.Page.objects.get(id=page_id)
+def _get_pages():
+    """Delegate page-fetching based on the configuration; if THISDAY is
+    requested and gets valid pages, the second argument returned is True to
+    signal to the template which title to use"""
+
+    # We use random pages if explicitly requested *or* THISDAY is requested but
+    # no pages are found
+    isrand = config.RANDOM or config.THISDAY
+
+    if config.THISDAY:
+        pages = _pages_this_day()
+        if len(pages) > 0:
+            return pages, True
+
+    if isrand:
+        return _random_pages(config.NUMBER), False
+
+    return map(_get_page_by_info, config.PAGES), False
+
+def _pages_this_day():
+    """Find any pages within the min/max years and today's month/year"""
+    pages = []
+
+    # Filtering variables
+    dt_range_start = datetime.date(config.MINYEAR, 1, 1)
+    dt_range_end = datetime.date(config.MAXYEAR, 12, 31)
+    now = datetime.date.today()
+
+    # Grab each issue that matches, and create a page_info structure for the
+    # first page of that issue.  For sanity's sake, we only pull 100 max.
+    issues = models.Issue.objects
+    issues = issues.filter(date_issued__range=(dt_range_start, dt_range_end))
+    issues = issues.filter(date_issued__month = now.month)
+    issues = issues.filter(date_issued__day = now.day)
+    for issue in issues[:100]:
+        if first_page and first_page.jp2_filename:
+            pages.append({
+                'date': issue.date_issued,
+                'edition': issue.edition,
+                'lccn': issue.title.lccn,
+                'name': issue.title.name,
+                'page_obj': first_page,
+                'sequence': first_page.sequence,
+            })
+
+    return pages
+
+def _random_pages(limit):
+    page_len = models.Page.objects.count()
+    indices = random.sample(xrange(page_len), limit)
+    pages = []
+    for index in indices:
```

Comment:
I was just concerned it might still be causing it to run a new query each time this way and unsure which is more efficient. But if it's already much faster even for your larger volume, we should be good.